plugins/autoid/plugin.js

```diff
@@ -1,9 +1,95 @@
 'use strict';
 
 (function () {
-	CKEDITOR.plugins.add('autoid', {
-		init: function (editor) {
+  var editorHasFocus = false;
+  var commandIsActive = false;
+  var EVENT_NAMES = {
+    ALL_IDS_COMPLETE: 'allIdsComplete',
+    ID_ADDED: 'idAdded'
+  };
 
-		}
-	});
+  CKEDITOR.plugins.add('autoid', {
+    init: function (editor) {
+      var self = this;
+
+      editor.addFeature({
+        allowedContent: 'h1 h2 h3 h4 h5 h6 *[id]'
```

Comment:
this might be worth commenting with an explanation.  with the many forms of CKEditor allowedContent it's pretty hard to remember to mentally parse this as "id is allowed for all the previous elements"